File: frontend/demo/component/numberfield/number-field-step-buttons.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/form-layout';
import '@vaadin/form-layout/vaadin-form-item.js';
import '@vaadin/integer-field';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('number-field-step-buttons')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <vaadin-form-layout auto-responsive labels-aside>
        <vaadin-form-item>
          <label slot="label">Adults</label>
          <!-- tag::snippet[] -->
          <vaadin-integer-field
            value="2"
            step-buttons-visible
            min="0"
            max="9"
          ></vaadin-integer-field>
          <!-- end::snippet[] -->
        </vaadin-form-item>

        <vaadin-form-item>
          <label slot="label">
            <div>Children</div>
            <div style="font-size: 0.75em; position: absolute">Age 2-12</div>
          </label>
          <vaadin-integer-field
            value="2"
            step-buttons-visible
            min="0"
            max="9"
          ></vaadin-integer-field>
        </vaadin-form-item>

        <vaadin-form-item>
          <label slot="label">Infants</label>
          <vaadin-integer-field
            value="1"
            step-buttons-visible
            min="0"
            max="9"
          ></vaadin-integer-field>
        </vaadin-form-item>
      </vaadin-form-layout>
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/numberfield/NumberFieldStepButtons.java
package com.vaadin.demo.component.numberfield;

import com.vaadin.flow.component.formlayout.FormLayout;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.textfield.IntegerField;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("number-field-step-buttons")
public class NumberFieldStepButtons extends FormLayout {

    public NumberFieldStepButtons() {
        setAutoResponsive(true);
        setLabelsAside(true);

        // tag::snippet[]
        IntegerField adultsField = new IntegerField();
        adultsField.setValue(2);
        adultsField.setStepButtonsVisible(true);
        adultsField.setMin(0);
        adultsField.setMax(9);
        // end::snippet[]
        addFormItem(adultsField, "Adults");

        IntegerField childrenField = new IntegerField();
        childrenField.setValue(2);
        childrenField.setStepButtonsVisible(true);
        childrenField.setMin(0);
        childrenField.setMax(9);

        Div children = new Div("Children");
        Div childrenExplainer = new Div();
        childrenExplainer.setText("Age 2-12");
        childrenExplainer.getStyle().set("font-size", "0.75em");
        childrenExplainer.getStyle().set("position", "absolute");

        addFormItem(childrenField, new Div(children, childrenExplainer));

        IntegerField infantsField = new IntegerField();
        infantsField.setValue(1);
        infantsField.setStepButtonsVisible(true);
        infantsField.setMin(0);
        infantsField.setMax(9);
        addFormItem(infantsField, "Infants");
    }

    public static class Exporter extends // hidden-source-line
            DemoExporter<NumberFieldStepButtons> { // hidden-source-line
    } // hidden-source-line
}
